R/tbl_hierarchical_rate_by_grade.R

```diff
@@ -0,0 +1,474 @@
+            label_grade = dplyr::case_when(
+              .data$variable == grade ~ label,
+              .data$variable == ae | label == "- Any adverse events -" ~ "- Any Grade -",
+              .default = ""
+            ),
+            .after = "label"
+          ) |>
+          # remove grade levels from label column
+          dplyr::mutate(label = if (.data$variable == grade) "" else label) |>
+          # remove statistics from summary rows if not an overall row
+          dplyr::mutate(
+            across(
+              all_stat_cols(),
+              ~ if (
+                .data$variable %in% c(ae, "..ard_hierarchical_overall..") |
+                  .data$label_grade %in% c(lvls, names(grade_groups)) |
+                  .data$label == "- Any adverse events -"
+              ) {
+                .
+              } else {
+                NA
+              }
+            )
+          ) |>
+          dplyr::ungroup()
+      }
+    ) |>
+    # show label_grade column
+    gtsummary::modify_column_unhide("label_grade") |>
+    gtsummary::modify_column_alignment("label_grade", align = "left") |>
+    # remove default footnote
+    gtsummary::remove_footnote_header(columns = everything()) |>
+    # convert "0 (0.0%)" to "0"
+    gtsummary::modify_post_fmt_fun(
+      fmt_fun = ~ ifelse(. == "0 (0.0%)", "0", .),
+      columns = all_stat_cols()
+    ) |>
+    # update header label
+    gtsummary::modify_header(
+      label = paste0("**", label[[soc]], "**  \n", paste0(rep("\U00A0", 4L), collapse = ""), "**", label[[ae]], "**"),
+      label_grade = paste0("\U00A0  \n**", label[[grade]], "**")
+    )
+
+  # indent grade level labels within grade groups
+  if (!is_empty(grade_groups)) {
+    tbl_final <- tbl_final |>
+      gtsummary::modify_indent(
+        columns = "label_grade", rows = .data$variable == grade & .data$label_grade %in% unlist(grade_groups),
+        indent = 4L
+      )
+  }
+
+  # return final table ---------------------------------------------------------
+  tbl_final$call_list <- list(tbl_hierarchical_rate_by_grade = match.call())
+  tbl_final$cards <- list(
+    tbl_hierarchical_rate_by_grade = list(tbl_hierarchical = tbl_final$cards$tbl_hierarchical)
+  )
+  tbl_final$inputs <- tbl_hierarchical_rate_by_grade_inputs
+
+  tbl_final |>
+    structure(class = c("tbl_hierarchical_rate_by_grade", "gtsummary"))
+}
+
+# function to generate the hierarchical ARD(s) by highest grade
+.ard_rate_by_grade_ordered <- function(data,
+                                       variables,
+                                       by,
+                                       id,
+                                       denominator,
+                                       grade_groups = NULL) {
+  # get name of grade variable
+  grade <- dplyr::last(variables)
+
+  if (!is_empty(grade_groups)) {
+    # if any grade groups present, replace grades with their grade groups
+    data[[grade]] <-
+      do.call(
+        forcats::fct_recode,
```

Comment:
Should this function be added to the forcats imports file (or a different function used)?

Reply:
I will use `forcats::fct_collapse()` instead and add it to the standalone file. Thanks!!